File: app/src/main/java/com/afterlogic/aurora/drive/presentation/modules/replace/view/ReplaceArgs.java
package com.afterlogic.aurora.drive.presentation.modules.replace.view;

import android.os.Bundle;

import com.afterlogic.aurora.drive.model.AuroraFile;
import com.afterlogic.aurora.drive.presentation.common.modules.v3.view.Args;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by aleksandrcikin on 04.07.17.
 * mail: mail@sunnydaydev.me
 */

public class ReplaceArgs extends Args {

    private static final String KEY_COPY_MODE = ReplaceArgs.class.getName() + ".copyMode";
    private static final String KEY_FILES = ReplaceArgs.class.getName() + ".files";

    public ReplaceArgs(Bundle args) {
        super(args);
    }

    public ReplaceArgs() {
        super();
    }

    public boolean isCopyMode() {
        return getBundle().getBoolean(KEY_COPY_MODE, false);
    }

    public List<AuroraFile> getFiles() {
        return getBundle().getParcelableArrayList(KEY_FILES);
    }

    public void setFiles(List<AuroraFile> files) {
        args.putParcelableArrayList(KEY_FILES, new ArrayList<>(files));
    }

    public void setCopyMode(boolean copyMode) {
        args.putBoolean(KEY_COPY_MODE, copyMode);
    }
}

File: app/src/main/java/com/afterlogic/aurora/drive/presentation/modules/replace/view/ReplaceArgs.kt
package com.afterlogic.aurora.drive.presentation.modules.replace.view

import android.os.Bundle

import com.afterlogic.aurora.drive.model.AuroraFile
import com.afterlogic.aurora.drive.presentation.common.modules.v3.view.Args

import java.util.ArrayList
import javax.inject.Inject

/**
 * Created by aleksandrcikin on 04.07.17.
 * mail: mail@sunnydaydev.me
 */

class ReplaceArgs : Args {

    var isCopyMode: Boolean
        get() = bundle.getBoolean(KEY_COPY_MODE, false)
        set(copyMode) = args.putBoolean(KEY_COPY_MODE, copyMode)

    var files: List<AuroraFile>
        get() = bundle.getParcelableArrayList(KEY_FILES)
        set(files) = args.putParcelableArrayList(KEY_FILES, ArrayList(files))

    constructor(args: Bundle) : super(args)

    constructor() : super()

    companion object {

        private val KEY_COPY_MODE = ReplaceArgs::class.java.name + ".copyMode"
        private val KEY_FILES = ReplaceArgs::class.java.name + ".files"
    }

    internal class Initializer @Inject constructor() {

        lateinit var args: ReplaceArgs

    }

}
